src/action/common/configure_init_service.rs

```diff
@@ -93,7 +93,7 @@ impl ConfigureInitService {
             InitSystem::Systemd => {
                 // If /run/systemd/system exists, we can be reasonably sure the machine is booted
                 // with systemd: https://www.freedesktop.org/software/systemd/man/sd_booted.html
-                if !(Path::new("/run/systemd/system").exists() || which::which("systemctl").is_ok())
+                if !(Path::new("/run/systemd/system").exists() && which::which("systemctl").is_ok())
                 {
                     return Err(Self::error(ActionErrorKind::SystemdMissing));
                 }
```

Comment:
```suggestion
                if !Path::new("/run/systemd/system").exists() {
                    return Err(Self::error(ActionErrorKind::SystemdMissing));
                }
                
                if which::which("systemctl").is_err() {
                    return Err(Self::error(ActionErrorKind::SystemdMissing));
                }
```

I know I wrote this condition initially, but trying to re-read it broke my brain a bit... Could I please ask you to switch to this more verbose set of checks which makes it more obvious what the error conditions are?

Reply:
Absolutely!